x/shield/keeper/proposal.go

```diff
@@ -220,57 +262,63 @@ func (k Keeper) UndelegateCoinsToShieldModule(ctx sdk.Context, delAddr sdk.AccAd
 
 	// start with bonded delegations
 	lossAmountDec := loss.ToDec()
-	for _, del := range delegations {
-		val, found := k.sk.GetValidator(ctx, del.GetValidatorAddr())
+	remainingDec := lossAmountDec
+	for i := range delegations {
+		val, found := k.sk.GetValidator(ctx, delegations[i].GetValidatorAddr())
 		if !found {
 			panic("validator is not found")
 		}
-		delAmountDec := val.TokensFromShares(del.GetShares())
+		delAmountDec := val.TokensFromShares(delegations[i].GetShares())
```

Comment:
My understanding is here we don't have to use index because we're not modifying value.

Reply:
According to Zhaozhong it's not a good coding style to have both `i` and `del`.